### ISSUE TITLE
Upgrade needed for JLink adapter to support asyncio updates

### ISSUE THREAD

**Issue by anwu** [iotile/coretools#802]
JLink adapter needs to be upgraded to support the recent asyncio updates.
Currently, the `iotile.core.hw.transport.adapter.async_wrapper` is giving the error 
```
Inside connect, conn_id=0, conn_string=device=nrf52
```

Here is the output of my console with the `master` version of `iotile-core` and `iotile-transport-jlink` being locally deployed.
```
(env_p36) PS C:\Arch> iotile -vvvvvv hw --port=jlink
19-04-25 10:42:33.402 DEB iotile.core.utilities.async_tools.event_loop Starting event loop
19-04-25 10:42:33.402 DEB asyncio Using selector: SelectSelector
19-04-25 10:42:33.404 DEB iotile.core.utilities.async_tools.event_loop Starting loop in background thread
19-04-25 10:42:33.404 DEB iotile.core.utilities.async_tools.event_loop Added primary task LegacyAdapter (JLinkAdapter)
19-04-25 10:42:33.404 DEB iotile.core.utilities.async_tools.event_loop Added subtask periodic task for JLinkAdapter to parent LegacyAdapter (JLinkAdapter)
(HardwareManager) connect_direct device=nrf52
19-04-25 10:42:40.208 INF iotile.core.hw.transport.adapter.async_wrapper Inside connect, conn_id=0, conn_string=device=nrf52
DeviceAdapterError: Operation none on conn None failed because connection id does not exist
Additional Information:
reason: connection id does not exist
operation: none
conn_id: None
```
*NOTE:* The `iotile-transport-jlink` looks like it works fine when the release version of `iotile-core` is installed with the `master` version of `iotile-transport-jlink`.

**Comment by mattrunchey** [comment]
Full stack trace from this command:
```
>>> hw.connect_direct(connection_string="device=nrf52")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "</Users/mrunchey/v.8/lib/python3.5/site-packages/decorator.py:decorator-gen-23>", line 2, in connect_direct
  File "/Users/mrunchey/v.8/lib/python3.5/site-packages/typedargs/utils.py", line 32, in _check_and_execute
    retval = func(*convargs, **convkw)
  File "/Users/mrunchey/gitrepos/coretools/iotilecore/iotile/core/hw/hwmanager.py", line 275, in connect_direct
    self.stream.connect_direct(connection_string)
  File "/Users/mrunchey/gitrepos/coretools/iotilecore/iotile/core/hw/transport/adapterstream.py", line 212, in connect_direct
    self._loop.run_coroutine(self.adapter.connect(0, connection_string))
  File "/Users/mrunchey/gitrepos/coretools/iotilecore/iotile/core/utilities/async_tools/event_loop.py", line 516, in run_coroutine
    return future.result()
  File "/usr/local/Cellar/python35/3.5.6_2/Frameworks/Python.framework/Versions/3.5/lib/python3.5/concurrent/futures/_base.py", line 405, in result
    return self.__get_result()
  File "/usr/local/Cellar/python35/3.5.6_2/Frameworks/Python.framework/Versions/3.5/lib/python3.5/concurrent/futures/_base.py", line 357, in __get_result
    raise self._exception
  File "/usr/local/Cellar/python35/3.5.6_2/Frameworks/Python.framework/Versions/3.5/lib/python3.5/asyncio/tasks.py", line 242, in _step
    result = coro.throw(exc)
  File "/Users/mrunchey/gitrepos/coretools/iotilecore/iotile/core/hw/transport/adapter/async_wrapper.py", line 139, in connect
    resp = await self._execute(self._adapter.connect_sync, conn_id, connection_string)
  File "/usr/local/Cellar/python35/3.5.6_2/Frameworks/Python.framework/Versions/3.5/lib/python3.5/asyncio/futures.py", line 381, in __iter__
    yield self  # This tells Task to wait for completion.
  File "/usr/local/Cellar/python35/3.5.6_2/Frameworks/Python.framework/Versions/3.5/lib/python3.5/asyncio/tasks.py", line 310, in _wakeup
    future.result()
  File "/usr/local/Cellar/python35/3.5.6_2/Frameworks/Python.framework/Versions/3.5/lib/python3.5/asyncio/futures.py", line 294, in result
    raise self._exception
  File "/usr/local/Cellar/python35/3.5.6_2/Frameworks/Python.framework/Versions/3.5/lib/python3.5/concurrent/futures/thread.py", line 55, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/Users/mrunchey/gitrepos/coretools/iotilecore/iotile/core/hw/transport/adapter/legacy.py", line 173, in connect_sync
    self.connect_async(connection_id, connection_string, connect_done)
  File "/Users/mrunchey/gitrepos/coretools/transport_plugins/jlink/iotile_transport_jlink/jlink.py", line 268, in connect_async
    self._try_connect(connection_string)
  File "/Users/mrunchey/gitrepos/coretools/transport_plugins/jlink/iotile_transport_jlink/jlink.py", line 136, in _try_connect
    self._trigger_callback('on_disconnect', self.id, self._connection_id)
  File "/Users/mrunchey/gitrepos/coretools/iotilecore/iotile/core/hw/transport/adapter/legacy.py", line 135, in _trigger_callback
    func(*args, **kwargs)
  File "/Users/mrunchey/gitrepos/coretools/iotilecore/iotile/core/hw/transport/adapter/async_wrapper.py", line 263, in _on_disconnect
    conn_string = adapter._get_property(conn_id, 'connection_string')
  File "/Users/mrunchey/gitrepos/coretools/iotilecore/iotile/core/hw/transport/adapter/mixin_conndata.py", line 35, in _get_property
    raise DeviceAdapterError(conn_id, 'none', 'connection id does not exist')
iotile.core.hw.exceptions.DeviceAdapterError: DeviceAdapterError: Operation none on conn None failed because connection id does not exist
Additional Information:
conn_id: None
reason: connection id does not exist
operation: none
```

**Comment by mattrunchey** [comment]
Was this fixed in #807  @anwu ?

**Comment by anwu** [comment]
> Was this fixed in #807 @anwu ?

Yes, I'll go ahead and close this out.
